proxy/http/HttpConfig.cc

```diff
@@ -149,6 +149,40 @@ http_config_cb(const char * /* name ATS_UNUSED */, RecDataT /* data_type ATS_UNU
   return 0;
 }
 
+void
+Enable_Config_Var(std::string_view const &name, bool (*cb)(const char *, RecDataT, RecData, void *), void *cookie)
+{
+  // Must use this indirection because the API requires a pure function, therefore no values can
+  // be bound in the lambda. Instead this is needed to pass in the data for both the lambda and
+  // the actual callback.
+  using Context = std::tuple<decltype(cb), void *>;
+
+  // To deal with process termination cleanup, store the context instances in a deque where
+  // tail insertion doesn't invalidate pointers. These persist until process shutdown.
+  static std::deque<Context> storage;
```

Comment:
Why is there a static?

Reply:
It is static so the context persists until the callback is invoked. Otherwise it would go out of scope.

It was moved to the HTPP Config layer (not the HTTP layer) because it is directly involved in HTTP configuration, particularly with regard to transaction overridable values. Any user of this function will be using it to handle HTTP configuration variables. For instance, it needs to call `http_config_cb` which is in "HttpConfig.cc". Putting it in librecords would create yet another circular dependency.